### PR TITLE
[User Schedule] Expand ParallelType enum support

### PR DESCRIPTION
**Pull request by rdspring1**
This PR expands the ParallelTypes supported by the `parallelize` functions.
It also renames some of them for easier readability.

| CPP | Python |
| ------------- | ------------- |
| BIDx | grid_x |
| BIDy | grid_y |
| BIDz | grid_z |
| TIDx | block_x |
| TIDy | block_y |
| TIDz | block_z |
| Bulk | tma |

### PR List:
1. #2282 
2. #2283 **<<<  YOU ARE HERE**
3. #2284
4. #2285
5. #2286
6. #2287
7. #2288

### CODE REVIEW

File: csrc/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -2819,7 +2830,7 @@ void initNvFuserPythonBindings(PyObject* module) {
       py::arg("mesh"));
   //! experimental API for multidevice support
   nvf_sched.def(
-      "_parallelize",
+      "parallelize",
```
Comment (jjsjann123) [inline csrc/python_frontend/python_bindings.cpp:2833]: API change, we'll have to bump version number here :cry:

File: csrc/python_frontend/python_bindings.cpp [hunk 2]
```diff
@@ -406,9 +406,20 @@ void initNvFuserPythonBindings(PyObject* module) {
...
+      .value("grid_z", ParallelType::BIDz)
+      .value("block_x", ParallelType::TIDx)
+      .value("block_y", ParallelType::TIDy)
+      .value("block_z", ParallelType::TIDz)
+      .value("mma", ParallelType::Mma)
+      .value("serial", ParallelType::Serial)
+      .value("tma", ParallelType::Bulk)
+      .value("unroll", ParallelType::Unroll)
+      .value("unswitch", ParallelType::Unswitch)
+      .value("vectorize", ParallelType::Vectorize);
```
Comment (jjsjann123) [inline csrc/python_frontend/python_bindings.cpp:422]: I don't have a strong opinion on the name mapping. Let me advertise this on our channel to get opinion from a wider audience :stuck_out_tongue_winking_eye:

**Review by jjsjann123 — COMMENTED** [review]
LGTM. Let's bump version.txt since we are changing APIs.

**Comment by rdspring1** [comment]
!build

**Review by jjsjann123 — APPROVED** [review]
Thx for the version bump. we can discuss the naming of parallel type afterwards.

**Comment by rdspring1** [comment]
!build

**Pull request merged**